docs/docs/guides/18_migration_viem/index.md

```diff
@@ -4,4 +4,261 @@ title: 'Migration from Web3.js to Viem'
+console.log(wallet[0].address);
+```
+
+To migrate this to Viem, you'll use privateKeyToAccount() to create an account, and then can pass it to createWalletClient() for using it with client.
+
+```javascript
+import { createWalletClient, http } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
+import { mainnet } from 'viem/chains';
+
+const account = privateKeyToAccount(privateKey);
+
+const client = createWalletClient({
+	account,
+	chain: mainnet,
+	transport: http(),
+});
+```
+
+## Signing Messages
+
+Update how you handle message signing, following is web3.js example:
+
+```javascript
+const signature = web3.eth.accounts.sign('Some data', privateKey).signature;
+console.log(signature);
+```
+
+To sign message using Viem, you can use `signMessage()` method.
+
+```javascript
+import { createWalletClient, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+
+const walletClient = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!),
+});
+
+const [account] = await walletClient.getAddresses();
+
+const signature = await walletClient.signMessage({
+  account,
+  message: 'Some data',
+});
+
+```
+
+## Transaction
+
+When migrating transaction sending code, you'll need to update how transactions are signed and sent.
+
+```javascript
+const tx = await web3.eth.sendTransaction({
+	from: account,
+	to: '0x92d3267215Ec56542b985473E73C8417403B15ac',
+	value: web3.utils.toWei('0.001', 'ether'),
+});
+```
+
+In Viem there is `sendTransaction()` function avalible with walletClient.
+
+```javascript
+import { createWalletClient, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+
+const walletClient = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!),
+});
+
+const [account] = await walletClient.getAddresses();
+
+const hash = await walletClient.sendTransaction({
+  account,
+  to: '0x92d3267215Ec56542b985473E73C8417403B15ac',
+  value: 100000000000000n
+});
+
+```
+
+## Contracts
+
+### Contract Deployment
+
+When migrating contract deployment code, you'll need to update from Web3.js's deploy and send pattern:
+
+```javascript
+// use existing web3 instance connected with provider
+const contract = new web3.eth.Contract(abi);
+const deployTx = await contract
+	.deploy({
+		data: bytecode,
+		arguments: ['constructor param'],
+	})
+	.send({
+		from: account,
+		gas: '1000000',
+	});
+console.log(deployTx.options.address);
+```
+
+In Viem there is `deployContract()` function that can be used for contracts deployment.
+
+```javascript
+//import { deployContract } from 'viem'
+import { createWalletClient, custom } from 'viem';
+import { mainnet } from 'viem/chains';
+
+const walletClient = createWalletClient({
+	chain: mainnet,
+	transport: custom(window.ethereum),
+});
+
+const hash = await walletClient.deployContract({
+	abi,
+	account, //given account
+	args: [69420],
+	bytecode: '0x608060405260405161083e38038061083e833981016040819052610...',
```

Comment:
Should it be changed to
args: ['constructor param'],
bytecode: bytecode,

To make the snippet equal to what's in the web3js example?